core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -311,6 +311,22 @@ public ClientFactoryBuilder tlsCustomizer(Consumer<? super SslContextBuilder> tl
      * See <a href="https://tools.ietf.org/html/rfc7540#section-9.2.2">Section 9.2.2, RFC7540</a> for
      * more information. This option is disabled by default.
      */
+    public ClientFactoryBuilder tlsAllowUnsafeCiphers() {
+        return tlsAllowUnsafeCiphers(true);
+    }
+
+    /**
+     * Allows the bad cipher suites listed in
+     * <a href="https://tools.ietf.org/html/rfc7540#appendix-A">RFC7540</a> for TLS handshake.
+     *
+     * <p>Note that enabling this option increases the security risk of your connection.
+     * Use it only when you must communicate with a legacy system that does not support
+     * secure cipher suites.
+     * See <a href="https://tools.ietf.org/html/rfc7540#section-9.2.2">Section 9.2.2, RFC7540</a> for
+     * more information. This option is disabled by default.
+     *
+     * @param tlsAllowUnsafeCiphers Whether to allow the unsafe ciphers
+     */
     public ClientFactoryBuilder tlsAllowUnsafeCiphers(boolean tlsAllowUnsafeCiphers) {
```

Comment:
Maybe it's too late since this was already there but any thoughts on marking `@UnstableApi`? Should unsafe ciphers be supported forever? I know lots of tools are dropping support for them now.

Also, what do you think about logging a warning when it's enabled out of an abundance of caution?

Reply:
How about marking `@Deprecated` instead? I like the idea of logging a warning.

Reply:
Ah `@Deprecated` sounds even more precise yeah

Reply:
PTAL again :smile: